crates/shared/src/http_solver/model.rs

```diff
@@ -22,7 +22,10 @@ use crate::{
 #[derive(Clone, Debug, Default, Serialize)]
 pub struct BatchAuctionModel {
     pub tokens: BTreeMap<H160, TokenInfoModel>,
-    pub orders: BTreeMap<usize, OrderModel>,
+    /// Note that orders can contain 0x limit orders that are not sent to solvers as liquidity (`amms` field)
+    /// but as an order. Since 0x sets OrderUid externally (in their own format), these orders (Uid wise)
+    /// are not expected to be matched to orders from orderbook
+    pub orders: BTreeMap<OrderUid, OrderModel>,
```

Comment:
I don't think we should send an order-uid for non-CoW Protocol orders. I think thats a bit confusing. Can't we use a `uid: Option<OrderUid>` field on the `LimitOrder` instead?

Reply:
> Can't we use a uid: `Option<OrderUid>` field on the `LimitOrder` instead?

Not sure I follow this one. Do you suggest skipping serialization of the uid for `LimitOrder`s completely? I don't think this is possible because later when we receive solution we do some matching by ids.

Reply:
My suggestion is to, like for `amm`s, have the UID be a field in the limit order model that is passed (and keep IDs as-is)

Reply:
Hm, could do that, but having the ID as `key` takes less space and it feels like a proper way to go. Also, I plan to change the `key` for `amm`s to pool addreses instead of `usize` for the same reason.
Maybe changing it to the pub orders: `BTreeMap<String, OrderModel>` would be a good compromise, WDYT?

Reply:
> but having the ID as key takes less space and it feels like a proper way to go

I disagree, mostly because not all `LimitOrderModel`s have a valid UID - in fact we are "making up" UIDs for non-CoW Protocol order (i.e. 0x limit orders). To me, its an `Option<OrderUid>` because of this (and therefore, cannot be used as a key).

Reply:
https://github.com/cowprotocol/services/pull/721